app/policies/contest_policy.rb

```diff
@@ -31,9 +31,14 @@ def download_sources?
     user && user.admin?
   end
 
+  def toggle_runs_visible?
```

Comment:
Is there a reason to put this policy on the contest? I think it is more natural to be on the problem, as this is an action that is performed on the problem, not on the contest.